<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SnipetPropertyDialog</class>
 <widget class="QDialog" name="SnipetPropertyDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>573</width>
    <height>446</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Snipet property</string>
  </property>
  <layout class="QGridLayout">
   <item row="1" column="0" colspan="2">
    <widget class="QDialogButtonBox" name="m_buttons">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>&amp;Description</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="1">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>&amp;Description</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>&amp;Name :</string>
            </property>
            <property name="buddy">
             <cstring>m_nameLineEdit</cstring>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="m_nameLineEdit">
            <property name="whatsThis">
             <string>Name of the snipet used in to identify the snipet in menus.</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>&amp;Image :</string>
            </property>
            <property name="buddy">
             <cstring>m_iconComboBox</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="StringComboBox" name="m_iconComboBox"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>&amp;Extention :</string>
            </property>
            <property name="buddy">
             <cstring>m_extentionListWidget</cstring>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>&amp;Snipet key :</string>
            </property>
            <property name="buddy">
             <cstring>m_keyLineEdit</cstring>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="m_keyLineEdit">
            <property name="whatsThis">
             <string>Word key used to call the snipet in the &lt;b&gt;XINX&lt;/b&gt; editor. If no key is defined, it's not possible to call the template in the completion.</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>&amp;Automatic feature :</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QCheckBox" name="m_autoChk">
            <property name="text">
             <string>Call snipet automatically (not only by Ctrl+Space)</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QCheckBox" name="m_dialogChk">
            <property name="text">
             <string>Show description dialog, with parameters customization</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>&amp;Description :</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
            </property>
            <property name="buddy">
             <cstring>m_descriptionTextEdit</cstring>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QPlainTextEdit" name="m_descriptionTextEdit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>100</height>
             </size>
            </property>
            <property name="whatsThis">
             <string>Long descriptions of the snipet.</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="1">
           <widget class="QListWidget" name="m_extentionListWidget">
            <property name="selectionMode">
             <enum>QAbstractItemView::ExtendedSelection</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>&amp;Category</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Category</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0" colspan="3">
           <widget class="CategoryTreeView" name="m_categoryTreeView">
            <property name="rootIsDecorated">
             <bool>false</bool>
            </property>
            <attribute name="headerVisible">
             <bool>false</bool>
            </attribute>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="m_removeCategoryButton">
            <property name="text">
             <string>&amp;Remove Category</string>
            </property>
            <property name="icon">
             <iconset resource="../../components/xinxpluginselector.qrc">
              <normaloff>:/images/remove.png</normaloff>:/images/remove.png</iconset>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="m_addCategoryButton">
            <property name="text">
             <string>&amp;Add Category</string>
            </property>
            <property name="icon">
             <iconset resource="../../components/xinxpluginselector.qrc">
              <normaloff>:/images/add.png</normaloff>:/images/add.png</iconset>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>&amp;Snipet</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QGroupBox" name="m_templateGroupBox">
         <property name="title">
          <string>&amp;Template text :</string>
         </property>
         <layout class="QGridLayout">
          <item row="0" column="0">
           <widget class="XinxCodeEditAction" name="m_textEdit" native="true">
            <property name="whatsThis">
             <string>Templat to show to screen. In this template you can define parameter by used &lt;i&gt;%n&lt;/i&gt; keyword where &lt;i&gt;n&lt;/i&gt; is a number. More sophisticate snipet can use ECMAScript. Balise &lt;i&gt;&amp;lt;?= ?&amp;gt;&lt;/i&gt; is used to call script and show the text returned by the script. &lt;i&gt;&amp;lt;?    ?&amp;gt;&lt;/i&gt; can be used to init the script and make some pre-process.</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>&amp;Parameters</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QGroupBox" name="m_paramGroupBox">
         <property name="minimumSize">
          <size>
           <width>250</width>
           <height>0</height>
          </size>
         </property>
         <property name="whatsThis">
          <string>List of name parameter used by the template. The first parameter replace the key word %1, the second parameter replace the key word %2, ...</string>
         </property>
         <property name="title">
          <string>&amp;Parameter</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="0" column="0">
           <widget class="QTableView" name="m_parameterTable">
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_5">
      <attribute name="title">
       <string>&amp;Control</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>&amp;Condition to enable the snipet</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="0">
           <widget class="QPlainTextEdit" name="m_availablePlainTextEdit"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>XinxCodeEditAction</class>
   <extends>QWidget</extends>
   <header location="global">editors/xinxcodeedit.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>StringComboBox</class>
   <extends>QComboBox</extends>
   <header location="global">stringcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>CategoryTreeView</class>
   <extends>QTreeView</extends>
   <header>snipets/categorytreeview.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>m_nameLineEdit</tabstop>
  <tabstop>m_keyLineEdit</tabstop>
  <tabstop>m_descriptionTextEdit</tabstop>
  <tabstop>m_textEdit</tabstop>
 </tabstops>
 <resources>
  <include location="../../components/xinxpluginselector.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>m_buttons</sender>
   <signal>accepted()</signal>
   <receiver>SnipetPropertyDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>472</x>
     <y>429</y>
    </hint>
    <hint type="destinationlabel">
     <x>510</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_buttons</sender>
   <signal>rejected()</signal>
   <receiver>SnipetPropertyDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>326</x>
     <y>429</y>
    </hint>
    <hint type="destinationlabel">
     <x>510</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
